appointment-svc/controllers/patient-appointment-controller.ts

```diff
@@ -0,0 +1,118 @@
+import { Request, Response, NextFunction, query } from 'express';
+import AppointmentRepository from '../service/db-service/AppointmentRepository';
+import {
+  IUpdateAppointmentProps,
+  AppointmentStatus,
+} from '../types/appointment.interface';
+
+class PatientAppointmentController {
+  async rescheduleAppointment(req: Request, res: Response, next: NextFunction) {
+    const { decryptedData } = req.body;
+
+    const current_appointment_id = decryptedData.current_app_id;
+    const open_appointment_id = decryptedData.open_app_id;
+
+    const currentStatus = await AppointmentRepository.checkStatus(
+      open_appointment_id
+    );
+
+    if (currentStatus === 'BOOKED' || currentStatus === 'CONFIRMED') {
```

Comment:
I assume that this comparison would still work with AppointmentStatus enum, but for cleanliness, the enum would work better